### DISCUSSION TITLE
Backend considerations

### DISCUSSION THREAD

**Discussion by ghost** [xocomil/AngularSudoku#39] [Ideas]
It would be great to also develop a back-end API and database from scratch. ie. without using firebase or similar offerings.

I would be inclined to use NestJs, Postgres, Prisma, GraphQL, Rest etc.

The Api could facilitate features such user management, saving and sharing of results, times, etc.

i would also be inclined to create plugin generators for libraries as we go along.  So when we create our first @nrwl/angular library for a feature, i would create a plugin... 'angular-feature' this plugin would ensure tags are entered, configure routing (it's a feature) and any other options. maybe it requires a component store provider...

the same with a UI (dumb) component. same, without routing and store but specifying tags and any other domain requirements.

the generator could also add default stories and tests configured with any dependancies that were sent to the generator args

**Comment by xocomil** [comment]
I like this idea. I would like to potentially do a NestJS backend that could facilitate "multiplayer" type games. I really like where your mind is on this. I think all of these ideas are excellent.

**Reply by ghost** [reply]
A workspace plugin generator to create NestJs libraries that is configured to use @swc/jest could be a good place to start as i don't think Nx provides the option out of the box.

here's some info about why we would want to do this....

[https://wallabyjs.com/blog/optimizing-typescript.html](url)

I haven't quite worked out yet how to configure Angular jest to use it..
